mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from collections import OrderedDict, Counter
+
+
+def from_tsv(fname, dtypes=None):
+    """Read a tsv file into an OrderedDict
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file being loaded.
+    dtypes : list
+        List of types to case the values loaded as. This is specified column by
+        column.
+        Defaults to None. In this case all the data is loaded as strings."""
+    data = np.loadtxt(fname, dtype=str, delimiter='\t')
+    column_names = data[0, :]
+    info = data[1:, :]
+    data_dict = OrderedDict()
+    if isinstance(dtypes, (list, tuple)):
+        if len(dtypes) == info.shape[1]:
+            for i, name in enumerate(column_names):
+                data_dict[name] = info[:, i].astype(dtypes[i]).tolist()
+    else:
+        for i, name in enumerate(column_names):
+            data_dict[name] = info[:, i].tolist()
+    return data_dict
+
+
+def to_tsv(data, fname):
+    """ Write an OrderedDict into a tsv file """
+    rows = len(data[list(data.keys())[0]])
+    with open(fname, 'w') as f:
```

Comment:
I would first gather everything into a string. And then open the file. It avoids and extra level of nesting

Reply:
I disagree that writing the whole string first is nicer. I think it personally is nicer to have the for loop within the context manager, but if you really think that it would be better to change it to produce the wring first then write it I'll change it.

Reply:
yeah, I only though of that just after I made the commit. Next one will have that :)